dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -197,14 +204,41 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
             try
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                context = await f.InvokeAsync(context, cancellationToken: cancellationToken).ConfigureAwait(false);
+                var functionDetails = f.Describe();
+                string? renderedPrompt = null;
+
+                SemanticFunction? semanticFunction = f as SemanticFunction;
+
+                if (semanticFunction is not null)
+                {
+                    semanticFunction.AddDefaultValues(context.Variables);
+                    renderedPrompt = await semanticFunction.RenderPromptTemplateAsync(context, cancellationToken).ConfigureAwait(false);
+                }
+
+                var cancelled = this.OnFunctionInvoking(functionDetails, context, renderedPrompt);
```

Comment:
```suggestion
                SKContext promptContext = this.CreateNewContext();
                promptContext.Update(renderedPrompt);
                var cancelled = this.OnFunctionInvoking(functionDetails, promptContext);
```
...then we don't need `string? Prompt` to be a hard-coded property on the EventArgs classes.